### PR TITLE
 Improve logging in ws-server

### PR DESCRIPTION
**Pull request by Ivo-Yankov**
**Description**:
- Refactors existing logging and adds more details.
- Refactors the connection_limit logic so that it is easier to handle on the client side.

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by Nana-EC — APPROVED** [review]

File: packages/ws-server/src/webSocketServer.ts [hunk 1]
```diff
@@ -104,7 +102,7 @@ async function validateSubscribeEthLogsParams(filters: any, requestId: string) {
 
 app.ws.use(async (ctx) => {
     ctx.websocket.id = relay.subs()?.generateId();
-    logger.info(`New connection ${ctx.websocket.id}`);
+    logger.info(`New connection established. ConnectionId: ${ctx.websocket.id}. Current active connections: ${ctx.app.server._connections}`);
```
Comment (AlfredoG87) [inline packages/ws-server/src/webSocketServer.ts:105]: do we want to log all these at `info` level?

File: packages/ws-server/src/ConnectionLimiter.ts [hunk 2]
```diff
@@ -66,13 +65,20 @@ export default class ConnectionLimiter {
...
+        }
+
         const {ip} = ctx.request;
 
         // Limit connections from a single IP address
         const limitPerIp = parseInt(process.env.WS_CONNECTION_LIMIT_PER_IP || '10');
         if (this.clientIps[ip] && this.clientIps[ip] > limitPerIp) {
-            this.logger.info(`Maximum allowed connections from a single IP (${this.clientIps[ip]}) exceeded for address ${ip}`);
-            ctx.websocket.close(IP_LIMIT_ERROR.code, IP_LIMIT_ERROR.message);
+            this.logger.info(`Closing connection ${ctx.websocket.id} due to exceeded maximum connections from a single IP (${this.clientIps[ip]}) for address ${ip}`);
```
Comment (AlfredoG87) [inline packages/ws-server/src/ConnectionLimiter.ts:80]: `info` level is intented here or debug or trace as must of the code?

File: packages/ws-server/src/ConnectionLimiter.ts [hunk 3]
```diff
@@ -66,13 +65,20 @@ export default class ConnectionLimiter {
     }
 
     public applyLimits(ctx) {
+        // Limit total connections
+        if (this.connectedClients > parseInt(process.env.WS_CONNECTION_LIMIT || '10')) {
+            this.logger.info(`Closing connection ${ctx.websocket.id} due to exceeded maximum connections (${process.env.WS_CONNECTION_LIMIT})`);
```
Comment (AlfredoG87) [inline packages/ws-server/src/ConnectionLimiter.ts:70]: also, do we want `info` level here?

File: packages/relay/src/lib/subscriptionController.ts [hunk 4]
```diff
@@ -93,26 +92,26 @@ export class SubscriptionController {
         const {id} = connection;
 
         if (subId) {
-            this.logger.info(`${LOGGER_PREFIX} Unsubscribing connection ${id} from subscription ${subId}`);
+            this.logger.info(`Connection ${id}: Unsubscribing from ${subId}`);
         }
         else {
-            this.logger.info(`${LOGGER_PREFIX} Unsubscribing all instances of connection ${id}`);
+            this.logger.info(`Connection ${id}: Unsubscribing from all subscriptions`);
```
Comment (AlfredoG87) [inline packages/relay/src/lib/subscriptionController.ts:98]: do we want to log this as `info` level?

File: packages/relay/src/lib/subscriptionController.ts [hunk 5]
```diff
@@ -93,26 +92,26 @@ export class SubscriptionController {
         const {id} = connection;
 
         if (subId) {
-            this.logger.info(`${LOGGER_PREFIX} Unsubscribing connection ${id} from subscription ${subId}`);
+            this.logger.info(`Connection ${id}: Unsubscribing from ${subId}`);
```
Comment (AlfredoG87) [inline packages/relay/src/lib/subscriptionController.ts:95]: do we want to log this as info level?

File: packages/relay/src/lib/subscriptionController.ts [hunk 6]
```diff
@@ -71,13 +70,13 @@ export class SubscriptionController {
...
         if (existingSub) {
-            this.logger.debug(`${LOGGER_PREFIX} Connection ${connection.id} already subscribed to ${tag}`);
+            this.logger.debug(`Connection ${connection.id}: Attempting to subscribe to ${tag}; already subscribed`);
             return existingSub.subscriptionId;
         }
 
         const subId = this.generateId();
 
-        this.logger.info(`${LOGGER_PREFIX} New subscription ${subId}, listening for ${tag}`);
+        this.logger.info(`Connection ${connection.id}: created subscription ${subId}, listening for ${tag}`);
```
Comment (AlfredoG87) [inline packages/relay/src/lib/subscriptionController.ts:79]: do we want to log this as info level?

File: packages/relay/src/lib/poller.ts [hunk 7]
```diff
@@ -109,7 +114,7 @@ export class Poller {
     }
 
     remove(tag: string) {
-        this.logger.info(`${LOGGER_PREFIX} No longer polling for ${tag}`);
+        this.logger.info(`${LOGGER_PREFIX} Tag ${tag} removed from polling list`);
```
Comment (AlfredoG87) [inline packages/relay/src/lib/poller.ts:117]: do we want to log this as info level?

File: packages/relay/src/lib/poller.ts [hunk 8]
```diff
@@ -96,7 +101,7 @@ export class Poller {
 
     async add(tag: string, callback: Function) {
         if (!this.hasPoll(tag)) {
-            this.logger.info(`${LOGGER_PREFIX} Polling for ${tag}`);
+            this.logger.info(`${LOGGER_PREFIX} Tag ${tag} added to polling list`);
```
Comment (AlfredoG87) [inline packages/relay/src/lib/poller.ts:104]: do we want to log this as info level?

**Review by AlfredoG87 — COMMENTED** [review]
LGTM, only nits regarding log `info` level.

File: packages/ws-server/src/webSocketServer.ts (hunk 1 above)
Comment (Ivo-Yankov) [reply packages/ws-server/src/webSocketServer.ts:105]: My reasoning is the following - all logs that indicate the occurance of an event should be `info` and all logs that contain more detailed data should be `debug` and `trace`. By changing this env `LOG_LEVEL=info` we can filter out a lot of the clutter in the logs. I found this really useful when developing and debugging localy. Why do you think that those logs should be with `debug` level?

**Review by AlfredoG87 — APPROVED** [review]
LGTM 👍

**Review by Nana-EC — APPROVED** [review]
LG

**Pull request merged**